### PR TITLE
expose ImageData constructor

### PR DESCRIPTION
**Pull request by targos**
I need the ImageData constructor and I currently have to get it `with require('canvas/lib/bindings').ImageData`

Example use: convert a pixel array to dataURL
```js
function toDataURL(width, height, pixels) {
    var imgData = new ImageData(pixels, width, height);
    var canvas = new Canvas(width, height);
    var ctx = canvas.getContext('2d');
    ctx.putImageData(imgData, 0, 0);
    return canvas.toDataURL();
}
```

### CODE REVIEW

**Comment by kangax** [comment]
+1 from me, but we should add it to documentation too then.

**Comment by targos** [comment]
I will investigate further because it only works with the internal CanvasPixelArray objects.
`new ImageData(new Uint8ClampedArray(4), 1, 1)` fails

**Comment by eqyiel** [comment]
I also need the ImageData constructor.

The line @targos posted not fail since #604 (now that ImageData is using Uint8ClampedArray).  Can we get this merged?

**Comment by LinusU** [comment]
@targos Could you rebase on master? Also, adding documentation and tests will greatly increase the chances of this getting merged.

**Comment by targos** [comment]
I just found that the constructor does not behave correctly. For example, `new ImageData(2, 2)` creates an array with 4 elements when it should be 16 (width * height * 4)
I will try to fix that and add tests.

**Comment by zbjornson** [comment]
@Targos my mistake (never tested those since they weren't exposed). Let me know if you want help or want me to fix it. I'm going to change that area soon for #637.

**Comment by targos** [comment]
I pushed an update to ImageData.cc + tests. PTAL
For the error messages, I copied the output from Chrome.

**Comment by zbjornson** [comment]
Looks good!

File: src/ImageData.cc [hunk 1]
```diff
@@ -44,46 +44,64 @@ NAN_METHOD(ImageData::New) {
...
-#if NODE_MAJOR_VERSION == 0 && NODE_MINOR_VERSION <= 10
-      height = clampedArray->GetIndexedPropertiesExternalArrayDataLength() / width;
-#else
-      height = clampedArray->Length() / width;
-#endif
+    int size = length / 4;
+    if (width == 0) {
+      Nan::ThrowRangeError("The source width is zero.");
+      return;
+    } else if (size % width != 0) {
```
Comment (LinusU) [inline src/ImageData.cc]: You don't need `else` here, please change to

```c++
}
if (size % width != 0) {
```

**Comment by LinusU** [comment]
One small nit and if you could please rebase on master again (sorry), we have updated the travis config and it will ensure that your tests runs with our new config. Thank you!

**Comment by targos** [comment]
@LinusU done :)

**Comment by LinusU** [comment]
Cool! Merging now :+1:

**Comment by LinusU** [comment]
Released as 1.3.0 \o/

**Comment by targos** [comment]
Thank you!

**Pull request merged**